quarkus-cli/src/test/java/io/quarkus/ts/quarkus/cli/QuarkusCliSpecialCharsIT.java

```diff
@@ -0,0 +1,104 @@
+package io.quarkus.ts.quarkus.cli;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.List;
+
+import javax.inject.Inject;
+
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.condition.EnabledIfSystemProperty;
+
+import io.quarkus.test.bootstrap.QuarkusCliClient;
+import io.quarkus.test.scenarios.QuarkusScenario;
+import io.quarkus.test.scenarios.annotations.DisabledOnQuarkusVersion;
+import io.quarkus.test.utils.FileUtils;
+
+@Tag("quarkus-cli")
+@QuarkusScenario
+@DisabledOnQuarkusVersion(version = "1\\..*", reason = "Quarkus CLI has been reworked in 2.x")
+public class QuarkusCliSpecialCharsIT {
+
+    static final String ARTIFACT_ID = "app";
+    static final String EXPECTED_BUILD_OUTPUT = "BUILD SUCCESS";
+    static final Path TARGET = Path.of("target");
+
+    static final List<String> SPECIAL_CHARS_FOLDERS = Arrays.asList(
+            // With spaces
+            "s p a c e s",
+            // With special chars
+            ",;~!@#$%^&()",
+            // With diacritics
+            "ěščřžýáíéůú",
+            // With japanese
+            "元気かい",
+            // With others
+            "Îñţérñåţîöñåļîžåţîờñ");
+
+    @Inject
+    static QuarkusCliClient cliClient;
+
+    private QuarkusCliClient.Result result;
+
+    @Test
+    public void shouldCreateApplicationOnJvmWithSpaces() {
+        for (String folder : SPECIAL_CHARS_FOLDERS) {
```

Comment:
Another approach could be using a `@ParametrizedTest` and `@ArgumentsSource`.

Reply:
I prefer individual test for the string like in https://github.com/quarkus-qe/quarkus-startstop/blob/main/testsuite/src/it/java/io/quarkus/ts/startstop/SpecialCharsTest.java#L159

Reply:
Changed